Driverless AI/scoring-pipeline-deployment-to-apache-nifi/scoring-pipeline-deployment-to-apache-nifi.md

```diff
@@ -0,0 +1,352 @@
+
+### Add Custom NiFi Processor to NiFi Product
+
+Once you have compiled the NiFi custom processor, a NAR file will be generated. A NAR allows several components and their dependencies to be packaged together into a single package. A NAR package is provided ClassLoader isolation from other NAR packages. So when software from many different organizations is all hosted within the same environment, the Java ClassLoaders will not be an issue.
+
+### Build a NiFi Flow to do Interactive or Batch Scoring
+You could build a NiFi flow from scratch with the custom processor **ExecuteDaiMojoScoringPipeline** to do interactive scoring or batch scoring. You would need at least a processor to ingest the source of data, the MOJO processor to perform predictions on the data and another processor to send the predicted results to a destination.
+
+To make it easier to see how to run the MOJO in the NiFi flow, we will use NiFi flow templates. A NiFi flow template is a flow that has already been built to ingest data, perform transformations on the data and send the data somewhere. There are two NiFi flow templates that have been created: one that can do interactive (real-time) scoring and the other one that can do batch scoring. It just depends on the data you are ingesting into the flow. If you ingest tabular data that has only 1 row of data in each file, then the **ExecuteDaiMojoScoringPipeline** will do interactive scoring. If you ingest tabular data that has multiple rows of data in each file, then the **ExecuteDaiMojoScoringPipeline** will do batch scoring. However, behind the scenes, even if the tabular data contains multiple rows, the NiFi processor iterates over each individual row and performs the prediction on each row.
+
+For reference, here is the path the two NiFi flow templates:
```

Comment:
rewrite this sentence: "For reference, here is the path the two NiFi flow templates:"

to this: "For reference, here is the path to the two NiFi flow templates:"